micropython/micropython.py

```diff
@@ -64,11 +66,17 @@ def is_test_with_regex_output(test_path: str):
     return p.parent.name + "/" + p.name in TESTS_WITH_REGEX_OUTPUT
 
 
-def get_test(dut):
-    dut.expect(PROMPT, timeout=45)
-    output = dut.before
+def remove_ansi_escape(text: str):
+    ansi_escape = re.compile(r"\x1B(?:[@-Z\\-_]|\[[0-?]*[ -/]*[@-~])")
+    return ansi_escape.sub("", text)
+
+
+def get_test(dut: Dut, ctx: TestContext):
+    dut.expect_exact(ctx.target.shell_prompt, timeout=10)
+    output = remove_ansi_escape(dut.before)
```

Comment:
Why those escape sequences are so problematic?


Did you mean ascii?
```suggestion
    output = remove_ascii_escape(dut.before)
```